docs/source/topics/calculations/usage.rst

```diff
@@ -312,38 +312,169 @@ Note that the source path can point to a directory, in which case its contents w
+The following example shows that by specifying ``3``, the exact folder structure is kept:
+
+.. code:: bash
+
+    retrieve_list = [('path/sub/file_c.txt', '.', 3)]
+
+    └─ path
+        └─ sub
+           └─ file_c.txt
+
+For ``depth=2``, only two levels of nesting are kept (including the file itself) and so the ``path`` folder is discarded.
+
+.. code:: bash
+
+    retrieve_list = [('path/sub/file_c.txt', '.', 2)]
+
+    └─ sub
+       └─ file_c.txt
+
+The same applies for directories.
+By specifying a directory for the first element, all its contents will be retrieved.
+With ``depth=1``, only the first level ``sub`` is kept of the folder hierarchy.
+
+.. code:: bash
+
+    retrieve_list = [('path/sub', '.', 1)]
+
+    └── sub
+        ├─ file_c.txt
+        └─ file_d.txt
+
+
+Pattern matching
+................
+
+If the exact file or folder name is not known beforehand, glob patterns can be used.
+In the following examples, all files that match ``*c.txt`` in the directory ``path/sub`` will be retrieved.
+Since ``depth=0`` the files will be copied without the ``path/sub`` subdirectory.
+
+.. code:: bash
+
+    retrieve_list = [('path/sub/*c.txt', '.', 0)]
+
+    └─ file_c.txt
+
+To keep the subdirectory structure, one can set the depth parameter, just as in the previous examples.
+
+.. code:: bash
+
+    retrieve_list = [('path/sub/*c.txt', '.', 2)]
+
+    └── sub
+        └─ file_c.txt
+
+
+Specific target directory
+.........................
+
+The final folder hierarchy of the retrieved files in the retrieved folder is not only determined by the hierarchy of the remote working directory, but can also be controlled through the second and third elements of the instructions tuples.
+The final ``depth`` element controls what level of hierarchy of the source is maintained, where the second element specifies the base path in the retrieved folder into which the remote files should be retrieved.
+For example, to retrieve a nested file, maintaining the remote hierarchy and storing it locally in the ``target`` directory, one can do the following:
+
+.. code:: bash
+
+    retrieve_list = [('path/sub/file_c.txt', 'target', 3)]
+
+    └─ target
+        └─ path
+            └─ sub
+               └─ file_c.txt
+
+The same applies for folders that are to be retrieved:
+
+.. code:: bash
+
+    retrieve_list = [('path/sub', 'target', 1)]
+
+    └─ target
+        └── sub
+            ├─ file_c.txt
+            └─ file_d.txt
+
+The target relative path is also compatible with glob patterns in the source relative paths:
+
+.. code:: bash
 
-    calc_info.retrieve_list = [('relative/path/output/file_*[0-9].xml', '.', 1)]
+    retrieve_list = [('path/sub/*c.txt', 'target', 0)]
 
-The second item when using globbing *has* to be ``'.'`` and the depth works just as before.
-In this example, all files matching the globbing pattern will be copied in the directory ``output`` in the retrieved folder data node.
+    └─ target
+        └─ file_c.txt
```

Comment:
No, you would get the second. I feel this is clearly described by the examples in the "Specific target directory" subsection. It is not possible to rename the top level filename or folder name. Can you explain why you think this wasn't clear and what should be improved? I don't think adding the example you provide is any different from the example 
```
    retrieve_list = [('path/sub/file_c.txt', 'target', 3)]
    └─ target
        └─ path
            └─ sub
               └─ file_c.txt
```
that I give

Reply:
Well, I guess my logic was:

1. With the quoted example `[('path/sub/*c.txt', 'target', 0)]`, there might be several `*c.txt` files so renaming is not well defined.
2. With the example you mention `[('path/sub/file_c.txt', 'target', 3)]` the renaming is also not so well defined...like, at the same time you are asking to respect the structure but you want it to rename...the first path?
3. Actually the one I mention, `[('path/sub', 'target', 1)]`, is the one that might be the closest (even though is a folder instead of a file) but now in this case it says 1 instead of 0 and by how it looks it is very tempting to extrapolate and think if you made that change then you would get the following:

```

    retrieve_list = [('path/sub', 'target', 1)]
    └─ target
        └── sub
            ├─ file_c.txt
            └─ file_d.txt


    retrieve_list = [('path/sub', 'target', 0)]
    └─ target
        ├─ file_c.txt
        └─ file_d.txt
```

I know that you say in the docstring of the method that ```If ``depth`` equals ``0`` or ``1`` (they are equivalent)```, but this is not necessarily known by people going over the docs and also it might be just restricted to the specific case of copying structure without any renaming indicated. I don't know, maybe a full example is not necessary, but I would maybe add what you said to the text: "Note that the target folder is always created and the files are copied inside of it: it is not possible to rename the top level file or folder with this".

Reply:
The examples and steps in your logic that you give are all based on the assumption that the second element of the tuple is the target name of the retrieved file or folder, rather than the name of the folder into which it will be stored. Of course when you assume this, the rest doesn't make sense, but it simply isn't the case.

I clearly state this behavior in the docstring of the datastructures module:

> The ``source`` and ``target`` elements are relative filepaths in the remote and retrieved folder. The contents of ``source`` (whether it is a file or folder) are copied in its entirety to the ``target`` subdirectory in the retrieved folder.

But to be fair the documentation could be clarified. How about I change the explanation to the following:

```
* source relative path: the relative path, with respect to the working directory on the remote, of the file or directory to retrieve
* target relative path: the relative path of the directory into which to copy the contents indicated by `source relative path` in the retrieved folder. The string '.' indicates the top level in the retrieved folder.
* depth: the number of levels of nesting in the folder hierarchy of the source relative path to maintain when copying, starting from the deepest file
```

Would that solve the confusion?

Reply:
Ok, yeah so just to clarify, it is probably the case that if one reads carefully and thinks it through, most of the people would arrive to the correct current behavior. But I do think that that requires either an extreme detailed and unbiased way of reading or a very thorough post-processing and synthesis of the various parts of the text; I'm not sure this is the typical disposition we can expect of people when reading the documentation (it was not mine at least, haha). I feel like the assumption that "well this looks like its doing some sort of underlying cp so it shoulda- kinda- work like that" would be a fairly frequent one and we don't lose much by emphasizing in a couple of places that "please note that is not the case".

> But to be fair the documentation could be clarified. How about I change the explanation to the following:

Did you already made the change? I don't see where would it go, it already says this in the top of the doc section (line 325?). I don't know, I still get the feeling that this a very tempting line to read over and just say "yeah yeah relative target path gotcha"; you really need to lawyer-read very textually to catch this in passing. For me the best solution is still to add a disclaimer in the parts where people that read over this part would go to get the practical examples as a reminder that things don't work that way.

Anyways, also you wrote the section and I'm starting to feel a bit dense, so I will cut it out. If this is not bothersome to you and you want more details on what exactly would make it clearer for me I will answer, no problem; otherwise I'll wait till you ping me again once you decided what to do with this and are ready for me to approve.

Reply:
I will add a commit to try and address your comments and then ping you again. You can then see if those changes do accomplish what you had in mind